### PR TITLE
Added OWNCLOUD_REDIS_ vars for Redis clustering

### PR DESCRIPTION
**Pull request by galexrt**
This adds OWNCLOUD_REDIS_ vars for Redis clustered access configuration, as seen here: https://doc.owncloud.com/server/10.1/admin_manual/configuration/server/config_sample_php_parameters.html#connection-details-for-a-redis-cluster.

### CODE REVIEW

File: rootfs/etc/templates/config.php [hunk 1]
```diff
@@ -469,18 +469,38 @@ function getConfigFromEnv() {
...
-
-      if (getenv('OWNCLOUD_REDIS_DB') != '') {
-        $config['redis']['dbindex'] = getenv('OWNCLOUD_REDIS_DB');
-      }
-
-      if (getenv('OWNCLOUD_REDIS_PASSWORD') != '') {
-        $config['redis']['password'] = getenv('OWNCLOUD_REDIS_PASSWORD');
+      switch (true) {
+        case getenv('OWNCLOUD_REDIS_SEEDS') != '':
+          $config['redis']['seeds'] = explode(',', getenv('OWNCLOUD_REDIS_SEEDS'));
```
Comment (tboerger) [inline rootfs/etc/templates/config.php]: For clustered redis the key for to be redis.cluster.

Comment (galexrt) [reply rootfs/etc/templates/config.php]: Fixed.

File: rootfs/etc/templates/config.php [hunk 2]
```diff
@@ -469,22 +469,47 @@ function getConfigFromEnv() {
...
+      switch (true) {
+        case getenv('OWNCLOUD_REDIS_SEEDS') != '':
+          $config['redis.cluster']['seeds'] = explode(',', getenv('OWNCLOUD_REDIS_SEEDS'));
+
+          if (getenv('OWNCLOUD_REDIS_TIMEOUT') != '') {
+            $config['redis.cluster']['timeout'] = (float) getenv('OWNCLOUD_REDIS_TIMEOUT');
+          }
+
+          if (getenv('OWNCLOUD_REDIS_READ_TIMEOUT') != '') {
+            $config['redis.cluster']['read_timeout'] = getenv('OWNCLOUD_REDIS_READ_TIMEOUT');
```
Comment (tboerger) [inline rootfs/etc/templates/config.php]: I think you need to typecast it to float as well

File: rootfs/etc/templates/config.php [hunk 3]
```diff
@@ -469,22 +469,47 @@ function getConfigFromEnv() {
         'memcache.distributed' => '\OC\Memcache\Redis',
         'memcache.locking' => '\OC\Memcache\Redis',
 
-        'redis' => [
-          'host' => getenv('OWNCLOUD_REDIS_HOST'),
-          'port' => getenv('OWNCLOUD_REDIS_PORT'),
-        ],
+        'redis' => [],
```
Comment (tboerger) [inline rootfs/etc/templates/config.php]: You can drop this

File: rootfs/etc/templates/config.php [hunk 4]
```diff
@@ -469,22 +469,47 @@ function getConfigFromEnv() {
...
+        'redis.cluster' => [],
```
Comment (tboerger) [inline rootfs/etc/templates/config.php]: This can also be dropped.

**Comment by galexrt** [comment]
@tboerger Sorry for the delay on this, KubeCon and others completely "broke" my schedule.

I have updated the PR with your feedback.

**Comment by tboerger** [comment]
@galexrt could you cherry-pick the commit for the xenial and bionic branch, please?

**Comment by galexrt** [comment]
@tboerger Here you go:

* https://github.com/owncloud-docker/base/pull/90
* https://github.com/owncloud-docker/base/pull/91

**Pull request merged**